expression/builtin.go

```diff
@@ -479,6 +483,8 @@ type builtinFunc interface {
 	evalDuration(row chunk.Row) (val types.Duration, isNull bool, err error)
 	// evalJSON evaluates JSON representation of builtinFunc by given row.
 	evalJSON(row chunk.Row) (val types.BinaryJSON, isNull bool, err error)
+	// evalArray evaluates JSON representation of builtinFunc by given row.
+	evalArray(row chunk.Row) (val types.TypedArray, isNull bool, err error)
```

Comment:
I suggest we reuse the existing `evalJSON` instead of introducing a new `evalArray`.